src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTownHall.java

```diff
@@ -85,6 +93,86 @@ public BuildingEntry getBuildingRegistryEntry()
         return ModBuildings.townHall;
     }
 
+    @Override
+    public void onColonyTick(final IColony colony)
+    {
+        super.onColonyTick(colony);
+
+        List<IGlobalResearch> removes = new ArrayList<>();
+        for(IGlobalResearch research : MinecoloniesAPIProxy.getInstance().getGlobalResearchTree().getAutostartResearches())
+        {
+            if (!IGlobalResearchTree.getInstance().isResearchRequirementsFulfilled(research.getResearchRequirement(), colony))
+            {
+                continue;
+            }
+            Map<BlockPos, IBuilding> buildings = colony.getBuildingManager().getBuildings();
+            int level = 0;
+            for (Map.Entry<BlockPos, IBuilding> building : buildings.entrySet())
+            {
+                if (building.getValue().getSchematicName().equals(ModBuildings.UNIVERSITY_ID))
+                {
+                    if (building.getValue().getBuildingLevel() > level)
+                    {
+                        level = building.getValue().getBuildingLevel();
+                    }
+                }
+            }
+            if (level < research.getDepth())
+            {
+                continue;
+            }
+            boolean researchAlreadyRun = false;
+            for (ILocalResearch progressResearch : colony.getResearchManager().getResearchTree().getResearchInProgress())
+            {
+                if(progressResearch.getId().equals(research.getId()))
+                {
+                    researchAlreadyRun = true;
+                    break;
+                }
+            }
+            // Don't want to spam people about in-progress or already-completed research.  Because these might change within a world,
+            // we can't just save them or check against effects.
+            if(researchAlreadyRun || Boolean.TRUE.equals(colony.getResearchManager().getResearchTree().hasCompletedResearch(research.getId())))
+            {
+                removes.add(research);
+                continue;
+            }
+
+            // if research has item requirements, only notify player; we don't want to have items disappearing from inventories.
+            if (!research.getCostList().isEmpty())
+            {
+                for (PlayerEntity player : this.colony.getMessagePlayerEntities())
+                {
+                    player.sendMessage(new TranslationTextComponent(TranslationConstants.RESEARCH_AVAILABLE, research.getDesc()), player.getUniqueID());
+                }
+            }
+            // Otherwise, we can start the research without user intervention.
```

Comment:
Possible.  I think a lot of the intended use case involves places where the player doesn't have a user interface to click yet, either because they don't have the University unlocked or it's not part of the intended gameplay path at that point, but there's definitely a tradeoff between requiring players to run back and forth with every unlock, and it being kinda mysterious why some start themselves and others don't.